SingularityService/src/test/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilderTest.java

```diff
@@ -138,6 +141,40 @@ public void testJobUserPassedAsEnvironmentVariable() {
   }
 
   @Test
+  public void testEnvironmentVariableOverrides() {
+    Map<String, String> overrideVariables = new HashMap<>();
+    overrideVariables.put("HS_ACCESS_GROUP", "test");
+    overrideVariables.put("STARTED_BY_USER", "notTestUser");
+
+    SingularityRunNowRequest runNowRequest = new SingularityRunNowRequestBuilder()
+        .environmentVariables(overrideVariables)
+        .build();
+
+    final SingularityRequest request = new SingularityRequestBuilder("test", RequestType.WORKER)
+        .build();
+    final SingularityDeploy deploy = new SingularityDeployBuilder("test", "1")
+        .setCommand(Optional.of("/bin/echo hi"))
+        .setRunImmediately(Optional.of(runNowRequest))
```

Comment:
This particular field specifies to run immediately with these settings upon successful completion of a deploy. In this case it isn't actually getting transferred to your pending task